### PR TITLE
Improve ``PauliFeatureMap`` docs

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

The docstring was technically not correct as pointed out in https://ibm-quantumcomputing.slack.com/archives/C021END1BFZ/p1683098942650589.

### Details and comments

This PR reformulates the docstring and fixes some other inconsistencies. For example, this class allows to choose not only the number of qubit connections, as the paper and current docs suggest, but the precise entanglement format.

### CODE REVIEW

File: qiskit/circuit/library/data_preparation/pauli_feature_map.py [hunk 1]
```diff
@@ -27,29 +27,34 @@ class PauliFeatureMap(NLocal):
...
+        U_{\Phi(\vec{x})}=\exp\left(i\sum_{S \in \mathcal{I}}
+        \phi_S(\vec{x})\prod_{i\in S} P_i\right).
 
-    The circuit contains ``reps`` repetitions of this transformation.
-    The variable :math:`P_i \in \{ I, X, Y, Z \}` denotes the Pauli matrices.
-    The index :math:`S` describes connectivities between different qubits or datapoints:
-    :math:`S \in \{\binom{n}{k}\ combinations,\ k = 1,... n \}`. Per default the data-mapping
+    Here, :math:`S` is a set of qubit indices that describes the connections in the feature map,
+    :math:`\mathcal{I}` is a set containing all these index sets, and
+    :math:`P_i \in \{I, X, Y, Z\}. Per default the data-mapping
```
Comment (Cryoris) [inline qiskit/circuit/library/data_preparation/pauli_feature_map.py:39]:
```suggestion
    :math:`P_i \in \{I, X, Y, Z\}`. Per default the data-mapping
```

File: qiskit/circuit/library/data_preparation/pauli_feature_map.py [hunk 2]
```diff
@@ -27,29 +27,34 @@ class PauliFeatureMap(NLocal):
...
-            \end{cases}
+            x_i \text{ if } S = \{i\} \\
+            \prod_{j \in S} (\pi - x_j) \text{ if } |S| > 1
+            \end{cases}.
+
+    The possible connections can be set using the ``entanglement`` and ``paulis`` argument.
+    For example, for single-qubit :math:`Z` rotations and two-qubit :math:`YY` interactions
+    between all qubit pairs, we can set
+
+    .. code::python
```
Comment (Cryoris) [inline qiskit/circuit/library/data_preparation/pauli_feature_map.py:53]:
```suggestion
    between all qubit pairs, we can set::

```

File: qiskit/circuit/library/data_preparation/pauli_feature_map.py [hunk 3]
```diff
@@ -27,29 +27,33 @@ class PauliFeatureMap(NLocal):
...
 
         \phi_S(\vec{x}) = \begin{cases}
-            x_0 \text{ if } k = 1 \\
-            \prod_{j \in S} (\pi - x_j) \text{ otherwise }
-            \end{cases}
+            x_i \text{ if } S = \{i\} \\
+            \prod_{j \in S} (\pi - x_j) \text{ if } |S| > 1
+            \end{cases}.
+
+    The possible connections can be set using the ``entanglement`` and ``paulis`` argument.
```
Comment (1ucian0) [inline qiskit/circuit/library/data_preparation/pauli_feature_map.py:49]:
```suggestion
    The possible connections can be set using the ``entanglement`` and ``paulis`` arguments.
```

File: qiskit/circuit/library/data_preparation/pauli_feature_map.py [hunk 4]
```diff
@@ -99,8 +104,9 @@ class PauliFeatureMap(NLocal):
         OrderedDict([('cx', 39), ('rx', 36), ('u1', 21), ('h', 15), ('ry', 12), ('rz', 12)])
 
     References:
-        [1]: Havlicek et al. (2018), Supervised learning with quantum enhanced feature spaces.
-           `arXiv:1804.11326 <https://arxiv.org/abs/1804.11326>`_
+        [1] Havlicek et al. Supervised learning with quantum enhanced feature spaces.
```
Comment (1ucian0) [inline qiskit/circuit/library/data_preparation/pauli_feature_map.py:107]: Something looks add in the reference section:

Screenshot 2023-05-06 at 11 15 12

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
Thanks for the quick fix! Just two comments and we can merge.

File: qiskit/circuit/library/data_preparation/pauli_feature_map.py (hunk 4 above)
Comment (Cryoris) [reply qiskit/circuit/library/data_preparation/pauli_feature_map.py]: Thanks for checking the docs 😁 Should be better in 8d90474

File: qiskit/circuit/library/data_preparation/pauli_feature_map.py [hunk 5]
```diff
@@ -99,8 +104,9 @@ class PauliFeatureMap(NLocal):
         OrderedDict([('cx', 39), ('rx', 36), ('u1', 21), ('h', 15), ('ry', 12), ('rz', 12)])
 
     References:
```
Comment (1ucian0) [inline qiskit/circuit/library/data_preparation/pauli_feature_map.py:106]:
```suggestion
    References:

```

Comment (1ucian0) [inline qiskit/circuit/library/data_preparation/pauli_feature_map.py:106]:
```suggestion
    References:

```

**Review by 1ucian0 — APPROVED** [review]

**Pull request merged**